scripts/deploy-upgradable.js

```diff
@@ -74,74 +101,83 @@ async function deploy(env, chains, wallet, artifactPath, contractName, deployTo)
-                    `${chain.name} | Signer ${wallet.address} does not match contract owner ${owner} for chain ${chain.name} in info.`,
+        try {
+            if (deployTo.length > 0 && !deployTo.find((name) => chain.name === name)) continue;
+            const rpc = chain.rpc;
+            const provider = getDefaultProvider(rpc);
+            console.log(`Gas override for chain ${chain.name}:`, chain.gasOptions);
+
+            if (chain[contractName] && chain[contractName].address) {
+                const contract = getProxy(wallet.connect(provider), chain[contractName].address);
+                const owner = await contract.owner();
+                console.log(`Proxy already exists for ${chain.name}: ${contract.address}`);
+                console.log(`Existing implementation ${await contract.implementation()}`);
+                console.log(`Existing owner ${owner}`);
+
+                if (wallet.address !== owner) {
+                    throw new Error(`Signer ${wallet.address} does not match contract owner ${owner} for chain ${chain.name} in info.`);
+                }
+
+                const anwser = readlineSync.question(`Perform an upgrade for ${chain.name}? (y/n) `);
+                if (anwser !== 'y') continue;
+
+                const args = await getImplementationArgs(contractName, chain, wallet.connect(provider));
+                console.log(`Implementation args for chain ${chain.name}: ${args}`);
+
+                await upgradeUpgradable(
+                    chain[contractName].address,
+                    wallet.connect(provider),
+                    implementationJson,
+                    args,
+                    getUpgradeArgs(contractName, chain),
                 );
+
+                chain[contractName].implementation = await contract.implementation();
+
+                console.log(`${chain.name} | New Implementation for ${contractName} is at ${chain[contractName].implementation}`);
+                console.log(`${chain.name} | Upgraded.`);
+            } else {
+                const key = env.includes('devnet') ? `${contractName}-${env}` : contractName;
+                const setupArgs = getInitArgs(contractName, chain);
+                console.log(`Proxy setup args: ${setupArgs}`);
+                console.log(`Proxy deployment salt: '${key}'`);
+
+                const proxyAddress = await predictContractConstant(chain.constAddressDeployer, wallet.connect(provider), proxyJson, key);
+                console.log(`Proxy will be deployed to ${proxyAddress}. Does this match any existing deployments?`);
+                const anwser = readlineSync.question(`Proceed with deployment on ${chain.name}? (y/n) `);
+                if (anwser !== 'y') return;
+
+                const args = await getImplementationArgs(contractName, chain, wallet.connect(provider));
+                console.log(`Implementation args for chain ${chain.name}: ${args}`);
+
+                const contract = await deployUpgradable(
```

Comment:
As we discussed, deploying the GMPExpressService will need to use create3 to deploy. Once I finish adding that I can either create a PR on this PR or create a new PR on main if we want to merge this now.